src/pkgcheck/checks/codingstyle.py

```diff
@@ -903,3 +903,66 @@ def feed(self, pkg):
                         yield RedundantDodir(
                             cmd.group('cmd'), line=dodir.group('call'),
                             lineno=lineno - 1, pkg=pkg)
+
+
+class UnquotedVariable(results.LineResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class UnquotedVariablesCheck(Check):
+    """Scan ebuild for variables that should be quoted like D, FILESDIR, etc."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([UnquotedVariable])
+
+    message_commands = (
+        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
+    )
+    var_names = (
```

Comment:
Now that I think about it, maybe a weird example, but all the usages of `{R,B,I,}DEPEND` should also be always quoted.

Reply:
Well, technically you aren't supposed to be using them in the first place ;-). And when people do, they probably want them unquoted to do word splitting.

Reply:
OK, I see. Than scratch that weird idea.